autoload/ale/c.vim

```diff
@@ -5,8 +5,12 @@ call ale#Set('c_parse_makefile', 0)
 call ale#Set('c_parse_compile_commands', 0)
 let s:sep = has('win32') ? '\' : '/'
 
-" Set just so tests can override it.
-let g:__ale_c_project_filenames = ['.git/HEAD', 'configure', 'Makefile', 'CMakeLists.txt']
+let g:ale_c_project_filenames = get(g:, 'ale_c_project_filenames', [
+\   '.git/HEAD',
+\   'configure',
+\   'Makefile',
+\   'CMakeLists.txt',
```

Comment:
```suggestion
\   'CMakeLists.txt',
\   'compile_flags.txt',
```

Let's add this here too, and write the same in the documentation for the option, so we do a little bit better project detection by default. You may have to break the default over many lines if it's long. You can use the hidden characters `>` and `<` in documentation to write multi-line code blocks. Search with `/>` and you'll find an example somewhere in the documentation.